src/handler.ts

```diff
@@ -76,13 +77,13 @@ abstract class AbstractFileHandler implements IMessageFilter {
   }
 
   /**
-   * Get the current set of widgets managed by the handler.
+   * Get the active widget for the handler (can be null).
    *
    * #### Notes
    * This is a read-only property
    */
-  get widgets(): Widget[] {
-    return this._widgets.slice();
+  get activeWidget(): Widget {
```

Comment:
Why make this a phosphor property, just for the changed signal? In that case, I'd make it a regular property and just emit the changed signal yourself.